src/validation/statement.rs

```diff
@@ -263,30 +263,45 @@ fn validate_direct_access<T: AnnotationMap>(
     }
 }
 
+fn validate_condition<T>(validator: &mut Validator, context: &ValidationContext<T>, condition: &AstNode)
+where
+    T: AnnotationMap,
+{
+    if let Some(value) = get_literal_int_or_const_expr_value(condition, context) {
+        if value == 0 || value == 1 {
+            return;
+        }
+    }
+
+    let kind = context.annotations.get_type_or_void(condition, context.index);
+
+    if !kind.get_type_information().is_bool() {
+        let slice = get_datatype_name_or_slice(validator.context, kind);
+        let message = format!("Expected a boolean, got `{slice}`");
+        let diagnostic = Diagnostic::new(message).with_location(condition.get_location());
+
+        if kind.get_type_information().is_int() {
+            // We're a bit more lenient with integers, generating a warning instead of an error
+            validator.push_diagnostic(diagnostic.with_error_code("E096"));
+        } else {
+            // ...anything else is a hard error
+            validator.push_diagnostic(diagnostic.with_error_code("E094"));
```

Comment:
Both the `E094` and `E096` return the same **message** but different severities (i.e. error vs warning). Do we need to generate different messages as well or this is fine as is?

Reply:
I'm a fan of this, especially because it improves the clarity for integer values but I'm wondering if we're not too intrusive here i.e. will this be annoying for the user? Though one could also argue the warning is intrusive, idk. Anyhow, done.

Reply:
Well, if the user feels like we are telling them things they are already well aware of, they can just disable this warning.